### ISSUE TITLE
Higher-order function with ParamSpec doesn't correctly handle generic callback argument

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4371]
This is an extension to the genericTypes101.py unit test that uses ParamSpec. The non-ParamSpec version works fine.

```python
from typing import Callable, ParamSpec, TypeVar

_T = TypeVar("_T")
_T1 = TypeVar("_T1")
_U = TypeVar("_U")


def ident(x: _U) -> _U:
    return x

def func2(__cb: Callable[[_T1], _T], __arg0: _T1) -> _T:
    ...

x1_0 = func2(ident, "hi")
reveal_type(x1_0, expected_text="str") # This works fine

x1_1 = func2(ident, 1)
reveal_type(x1_1, expected_text="int") # This works fine


_P = ParamSpec("_P")
_R = TypeVar("_R")
def func3(__obj: Callable[_P, _R], *args: _P.args, **kwargs: _P.kwargs) -> _R:
    ...

x2_0 = func3(ident, "hi")
reveal_type(x2_0, expected_text="str") # This is broken

x2_1 = func3(ident, 1)
reveal_type(x2_1, expected_text="int") # This is broken
```

This came to my attention via a bug filed in the mypy issue tracker: https://github.com/python/mypy/issues/14337.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.286, which I just published. It will also be included in a future version of pylance.
